### PR TITLE
fix(dart): update `dio` exceptions handling 

### PR DESCRIPTION
**Pull request by aallam**
## 🧭 What and Why

Starting from version `5.2.0`, the HTTP client Dio has started throwing errors in `dart analyze` due to a migration from `DioError` to `DioException`. Any errors thrown during `dart analyze` will cause the release process to fail.

### Changes included:

- The error handling logic has been updated to use `DioException`.
- The dio version has been updated.
- The Melos scripts have been updated to run `dart analyze` and will now fail if any errors are found.

### CODE REVIEW

File: clients/algoliasearch-client-dart/melos.yaml [hunk 1]
```diff
@@ -24,16 +24,24 @@ scripts:
...
-  test:
-    exec: dart test
-    description: Run `dart test` in all packages
+  update:
+    exec: dart pub update
+    description: Run `dart pub update` in all packages
+
+  upgrade:
+    exec: dart pub update
+    description: Run `dart pub update` in all packages
```
Comment (Fluf22) [inline clients/algoliasearch-client-dart/melos.yaml:44]: Is this an alias or a typo?

File: clients/algoliasearch-client-dart/melos.yaml [hunk 2]
```diff
@@ -24,16 +24,24 @@ scripts:
...
 
-  test:
-    exec: dart test
-    description: Run `dart test` in all packages
+  update:
+    exec: dart pub update
+    description: Run `dart pub update` in all packages
+
+  upgrade:
+    exec: dart pub update
```
Comment (millotp) [inline clients/algoliasearch-client-dart/melos.yaml:43]: should this say `upgrade` ? same for description

**Review by millotp — COMMENTED** [review]
nice

File: clients/algoliasearch-client-dart/melos.yaml (hunk 1 above)
Comment (aallam) [reply clients/algoliasearch-client-dart/melos.yaml:44]: a typo! nice catch!

File: clients/algoliasearch-client-dart/melos.yaml (hunk 2 above)
Comment (aallam) [reply clients/algoliasearch-client-dart/melos.yaml:43]: yup, that's a typo, should be "upgrade"

Comment (Fluf22) [reply clients/algoliasearch-client-dart/melos.yaml:43]: I'm faster 😎

**Review by Fluf22 — APPROVED** [review]
NIce 👌🏻

File: clients/algoliasearch-client-dart/melos.yaml (hunk 2 above)
Comment (millotp) [reply clients/algoliasearch-client-dart/melos.yaml]: it says 11h13 for both ahah, it's thanks to Darnetal 5G

**Pull request merged**